simulator/event_manager.go

```diff
@@ -370,25 +364,24 @@ func (c *EventHistoryCollector) SetCollectorPageSize(ctx *Context, req *types.Se
 		return body
 	}
 
-	ctx.WithLock(c.m, func() {
-		c.page = ring.New(size)
-		c.fillPage(size)
-	})
+	c.size = size
+	c.page = list.New()
+	ctx.WithLock(c.m, c.fillPage)
 
 	body.Res = new(types.SetCollectorPageSizeResponse)
 	return body
 }
 
 func (c *EventHistoryCollector) ResetCollector(ctx *Context, req *types.ResetCollector) soap.HasFault {
-	c.pos = c.m.history.Back()
+	c.pos = c.page.Back()
```

Comment:
Having a hard time understanding the intended behavior of this method reading the docs:

```
Moves the "scrollable view" to the item immediately preceding the "viewable latest page". 
If you use "readPrev", ReadPreviousTasks or ReadPreviousEvents, all items are retrieved from the 
newest item to the oldest item.
```

Is this line correct based on the above method description?

Reply:
Yes. When you call `ResetCollector()` pos is set to the back of the list, and new events are added to the back of the list.  After the Reset call, Read**Next**Events() will return `nil` until new events come in.
But you'll be able to read the entire history with `ReadPreviousEvents()` after calling Reset.

`RewindCollector()` is the opposite: Read**Prev**Events() will return `nil`, but can read the entire history with`ReadNextEvents()`

Reply:
Thx, I found the name `Reset` a bit confusing :)